hook.sh

```diff
@@ -279,18 +279,25 @@ while true; do
 
   display_warnings
 
-  # Ask the question (not using "read -p" as it uses stderr not stdout)
-  echo -en "${BLUE}Proceed with commit? [e/y/n/?] ${NC}"
+  if [ -t 1 ] || [ ! -z ${FAKE_TTY+x} ]; then
```

Comment:
Please can you provide some explanation as to what `[ ! -z ${FAKE_TTY+x} ]` is doing here?